repository parsing includes trailing slash when listing repository

### Description

A minor visual issue that occurs when the `origin` remote has a trailing `/` in the URL.

<img width="232" src="https://user-images.githubusercontent.com/359239/29757225-88565d5c-8bed-11e7-82f8-80cbaa1f3e36.png">

I can reproduce it in a failing test in [`remote-parsing-test.ts`](https://github.com/desktop/desktop/blob/master/app/test/unit/remote-parsing-test.ts):

```ts
  it('parses HTTPS URLs with a trailing slash ', () => {
    const remote = parseRemote('https://github.com/hubot/repo/')
    expect(remote).not.to.equal(null)
    expect(remote!.hostname).to.equal('github.com')
    expect(remote!.owner).to.equal('hubot')
    expect(remote!.name).to.equal('repo')
  })
```

Here's the test output:

```
...

  236 passing (32s)
  1 failing

  1) URL remote parsing parses HTTPS URLs with a trailing slash :

      AssertionError: expected 'repo/' to equal 'repo'
      + expected - actual

      -repo/
      +repo

      at Context.it (C:\Users\shiftkey\src\desktop\app\test\unit\remote-parsing-test.ts:11:29)
      at callFn (C:\Users\shiftkey\src\desktop\node_modules\mocha\lib\runnable.js:348:21)
      at Test.Runnable.run (C:\Users\shiftkey\src\desktop\node_modules\mocha\lib\runnable.js:340:7)
      at Runner.runTest (C:\Users\shiftkey\src\desktop\node_modules\mocha\lib\runner.js:443:10)
      at C:\Users\shiftkey\src\desktop\node_modules\mocha\lib\runner.js:549:12
      at next (C:\Users\shiftkey\src\desktop\node_modules\mocha\lib\runner.js:361:14)
      at C:\Users\shiftkey\src\desktop\node_modules\mocha\lib\runner.js:371:7
      at next (C:\Users\shiftkey\src\desktop\node_modules\mocha\lib\runner.js:295:14)
      at Immediate.<anonymous> (C:\Users\shiftkey\src\desktop\node_modules\mocha\lib\runner.js:339:5)
      at runCallback (timers.js:651:20)
      at tryOnImmediate (timers.js:624:5)
      at processImmediate [as _immediateCallback] (timers.js:596:5)
```

### Version

**GitHub Desktop version:** latest `master` e.g. 42b9453eb3b2091cf8f743a96d781b2113804b87

**OS version:** `Microsoft Windows [Version 10.0.15063]`

### Steps to Reproduce

 - Remove an existing repository from Desktop
 - Open this GitHub repository in the shell and rename the `origin` remote to drop the `.git` suffix and ensure a `/` exists on the folder, e.g.

```
$ git remote set-url origin https://github.com/shiftkey/octokat.js/
```
 - Add the repository to the app again

**Expected behavior:** repository name doesn't have a trailing `/`

**Actual behavior:** It totally has a trailing `/`

**Reproduces how often:** 100%

--- a/app/src/lib/remote-parsing.ts
+++ b/app/src/lib/remote-parsing.ts
@@ -23,7 +23,7 @@ export function parseRemote(url: string): IGitRemoteURL | null {
   // git@github.com:octocat/Hello-World.git
   // git:github.com/octocat/Hello-World.git
   const regexes = [
-    new RegExp('^https?://(?:.+@)?(.+)/(.+)/(.+?)(?:.git/?)?$'),
+    new RegExp('^https?://(?:.+@)?(.+)/(.+)/(.+?)(?:/|.git/?)?$'),
     new RegExp('^git@(.+):(.+)/(.+?)(?:.git)?$'),
     new RegExp('^git:(.+)/(.+)/(.+?)(?:.git)?$'),
   ]

--- a/app/test/unit/remote-parsing-test.ts
+++ b/app/test/unit/remote-parsing-test.ts
@@ -19,6 +19,14 @@ describe('URL remote parsing', () => {
     expect(remote!.name).to.equal('repo')
   })
 
+  it('parses HTTPS URLs with a trailing slash', () => {
+    const remote = parseRemote('https://github.com/hubot/repo/')
+    expect(remote).not.to.equal(null)
+    expect(remote!.hostname).to.equal('github.com')
+    expect(remote!.owner).to.equal('hubot')
+    expect(remote!.name).to.equal('repo')
+  })
+
   it('parses HTTPS URLs which include a username', () => {
     const remote = parseRemote('https://monalisa@github.com/hubot/repo.git')
     expect(remote).not.to.equal(null)

